diff --git a/hw/misc/Kconfig b/hw/misc/Kconfig
index fccd735c24..afe2716e17 100644
--- a/hw/misc/Kconfig
+++ b/hw/misc/Kconfig
@@ -24,7 +24,10 @@ config PCI_TESTDEV
     bool
     default y if TEST_DEVICES
     depends on PCI
-
+config GPU
+    bool
+    default y if TEST_DEVICES
+    depends on PCI && MSI_NONBROKEN
 config EDU
     bool
     default y if TEST_DEVICES
diff --git a/hw/misc/meson.build b/hw/misc/meson.build
index b1d8d8e5d2..feea70f9aa 100644
--- a/hw/misc/meson.build
+++ b/hw/misc/meson.build
@@ -1,4 +1,15 @@
 system_ss.add(when: 'CONFIG_APPLESMC', if_true: files('applesmc.c'))
+# Enable C++ for LLVM linking
+add_languages('cpp', native: false, required: true)
+
+llvm_dep = dependency('llvm',
+  version: '>=15',
+  method: 'config-tool',
+  modules: ['core', 'orcjit', 'native', 'executionengine'],
+  required: true
+)
+system_ss.add(when: 'CONFIG_GPU', if_true: [files('math3d.c', 'renderer.c', 'jit_decorators.c', 'jit_mem.c', 'jit_flow.c', 'jit_alu.c', 'jit.c', 'asm.c', 'gpu.c'), llvm_dep])
+
 system_ss.add(when: 'CONFIG_EDU', if_true: files('edu.c'))
 system_ss.add(when: 'CONFIG_FW_CFG_DMA', if_true: files('vmcoreinfo.c'))
 system_ss.add(when: 'CONFIG_ISA_DEBUG', if_true: files('debugexit.c'))
